lightning_client.go

```diff
@@ -1229,3 +1233,70 @@ func (s *lightningClient) ChannelBackups(ctx context.Context) ([]byte, error) {
 
 	return resp.MultiChanBackup.MultiChanBackup, nil
 }
+
+// PaymentRequest represents a request for payment from a node.
+type PaymentRequest struct {
+	// Destination is the node that this payment request pays to .
+	Destination string
+
+	// Hash is the payment hash associated with this payment
+	Hash lntypes.Hash
+
+	// Value is the value of the payment request in millisatoshis.
+	Value lnwire.MilliSatoshi
+
+	/// Timestamp of the payment request.
+	Timestamp time.Time
+
+	// Expiry is the time at which the payment request expires.
+	Expiry time.Time
+
+	// Description is a description attached to the payment request.
+	Description string
+
+	// PaymentAddress is the payment address associated with the invoice,
+	// set if the receiver supports mpp.
+	PaymentAddress []byte
+}
+
+// DecodePaymentRequest decodes a payment request.
+func (s *lightningClient) DecodePaymentRequest(ctx context.Context,
+	payReq string) (*PaymentRequest, error) {
+
+	rpcCtx, cancel := context.WithTimeout(ctx, rpcTimeout)
+	defer cancel()
+
+	rpcCtx = s.adminMac.WithMacaroonAuth(rpcCtx)
+
+	resp, err := s.client.DecodePayReq(rpcCtx, &lnrpc.PayReqString{
+		PayReq: payReq,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	hash, err := lntypes.MakeHashFromStr(resp.PaymentHash)
+	if err != nil {
+		return nil, err
+	}
+
+	paymentReq := &PaymentRequest{
+		Destination:    resp.Destination,
+		Hash:           hash,
+		Value:          lnwire.MilliSatoshi(resp.NumMsat),
+		Description:    resp.Description,
+		PaymentAddress: resp.PaymentAddr,
+	}
+
+	// Set our timestamp values if they are non-zero, because unix zero is
+	// different to a zero time struct.
+	if resp.Timestamp != 0 {
+		paymentReq.Timestamp = time.Unix(resp.Timestamp, 0)
+	}
+
+	if resp.Expiry != 0 {
```

Comment:
Isn't zero expiry equivalent to one hour?

Reply:
This should be handled lnd side (ie, we should never get 0 from lnd), and I don't want to start overlaying lnd-level logic here. 

If there's a bug in lnd, I think it's better to have expiry=0 than expiry= Sept 1970

Reply:
Looked this up quickly, fortunately it is handled on the LND side.